controllers/openstackdataplanerole_controller.go

```diff
@@ -125,6 +125,13 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 		return ctrl.Result{}, err
 	}
 
+	if instance.Status.Deployed && instance.DeletionTimestamp.IsZero() {
+		// The role is already deployed and not being deleted, so reconcileation
+		// is already complete.
+		logger.Info("Role already deployed", "instance", instance)
+		return ctrl.Result{}, nil
+	}
+
```

Comment:
This look ok. I'll test it and see if it addresses the issue. We would want this check much later though, as we want the other reconcile logic to still happen (inventory, ensureServices, etc).

Reply:
We should also likely set instance.Status.Deployed=false early within the "if instance.Spec.DeployStrategy.Deploy {" loop (~line 297)

Reply:
i belive that is the default but sure i can set this

Reply:
This still needs to be moved to probably just before the "if Deploy" loop.